src/workerd/server/workerd-api.c++

```diff
@@ -354,10 +372,47 @@ void WorkerdApi::compileModules(
   lockParam.withinHandleScope([&] {
     auto modules = jsg::ModuleRegistryImpl<JsgWorkerdIsolate_TypeWrapper>::from(lockParam);
 
+    if (util::Autogate::isEnabled(util::AutogateKey::BUILTIN_WASM_MODULES) &&
+        conf.getModules().size() > 0) {
+      auto mainModule = conf.getModules().begin();
+      // Inject pyodide bootstrap module.
+      {
+        capnp::MallocMessageBuilder message;
+        auto module = message.getRoot<config::Worker::Module>();
+        module.setEsModule(getPyodideBootstrap());
+        auto info = tryCompileModule(lockParam, module, modules->getObserver(), getFeatureFlags());
+        auto path = kj::Path::parse(mainModule->getName());
+        modules->add(path, kj::mv(KJ_REQUIRE_NONNULL(info)));
+      }
+      // Inject metadata that the bootstrap module will read.
+      auto metadataModuleName = kj::Path::parse("pyodide:current-bundle");
+      {
+        capnp::MallocMessageBuilder message;
+        auto module = message.getRoot<config::Worker::Module>();
+        module.setEsModule(generatePyodideMetadata(conf));
+
+        auto info = tryCompileModule(lockParam, module, modules->getObserver(), getFeatureFlags());
+        modules->add(metadataModuleName, kj::mv(KJ_REQUIRE_NONNULL(info)));
+      }
+      // Inject the pyodide-lock.json file.
+      auto lockModuleName = kj::Path::parse("pyodide:package-lock.json");
+      {
+        capnp::MallocMessageBuilder message;
+        auto module = message.getRoot<config::Worker::Module>();
+        module.setEsModule(getPyodideLock());
+
+        auto info = tryCompileModule(lockParam, module, modules->getObserver(), getFeatureFlags());
+        auto path = kj::Path::parse(mainModule->getName());
+        modules->add(lockModuleName, kj::mv(KJ_REQUIRE_NONNULL(info)));
+      }
+    }
+
     for (auto module: conf.getModules()) {
       auto path = kj::Path::parse(module.getName());
-      auto info = tryCompileModule(lockParam, module, modules->getObserver(), getFeatureFlags());
-      modules->add(path, kj::mv(KJ_REQUIRE_NONNULL(info)));
+      auto maybeInfo = tryCompileModule(lockParam, module, modules->getObserver(), getFeatureFlags());
+      KJ_IF_SOME(info, maybeInfo) {
```

Comment:
It looks like you've changed the meaning of `tryCompileModule()` returning `none`. It's a little weird because previously it never actually did return `none`, but for some reason it was defined to do so. In any case, the call site in `server.c++` needs to be updated to make sure it's doing the right thing under the new meaning.

Reply:
What would be the right thing to do under this new meaning? I assume just log an error. Updated to do that but let me know if that's right.

Reply:
for context... tryCompileModule returns a `kj::Maybe<T>` in support of the fallback service support that was recently added. The current implementation of the module registry does not currently return `kj::none` but it was done this way to prepare for another change I had started but decided to abandon. I think it's fine if the semantics are changed here. The fallback service callback returns kj::none if the module was not resolved by the fallback service.